### ISSUE TITLE
Document per client/group quotas

### ISSUE THREAD

**Issue by mattschumpert** [redpanda-data/documentation#1530]
We have a valuable feature (per client and per group of clients throughput quotas, with client identified by client_id) that's buried in details config docs: 

https://docs.redpanda.com/docs/reference/cluster-properties/#target_quota_byte_rate
https://docs.redpanda.com/docs/reference/cluster-properties/#kafka_client_group_byte_rate_quota

We need to explain these properly and how they work as they're an important part of defending clusters from offending tenants.  
We document the per IP stuff well, but not this, and even our PM/CEO didn't know we already have them :)

Per-IP doc is part of https://docs.redpanda.com/docs/manage/cluster-maintenance/configure-availability/#limit-client-connections

Perhaps this should be part of 'manage throughput': https://docs.redpanda.com/docs/manage/cluster-maintenance/manage-throughput/

Group of clients quota works like this:

> kafka_client_group_byte_rate_quota :
[{
  'group_name': 'first_group',
  'clients_prefix': 'group_1',
  'quota': 10240
}]>

@dlex is the core eng SME here

**Comment by shorgi** [comment]
@dlex is this blocked by https://github.com/redpanda-data/redpanda/issues/10251? Will the map config value be changed?

**Comment by dlex** [comment]
I don't think that's a stopper, after all `rpk` is not the only way to deal with cluster config.

**Comment by mattschumpert** [comment]
Sure would be nice to have it fixed though!

**Comment by mattschumpert** [comment]
@shorgi now that we have this documented we should clarify [here](https://docs.redpanda.com/docs/develop/kafka-clients/) we are still missing 'Per-user quotas for bandwidth and request rate, although Per-client-per-core and Per-client-group-per-core quotas are available: <New doc>' 

Right now it says we're missing per-client quotas which isn't totally true given the above.

**Comment by shorgi** [comment]
> @shorgi now that we have this documented we should clarify [here](https://docs.redpanda.com/docs/develop/kafka-clients/) we are still missing 'Per-user quotas for bandwidth and request rate, although Per-client-per-core and Per-client-group-per-core quotas are available: [Link to new doc]

@mattschumpert - created #1541 for this
